pangeo_forge_recipes/storage.py

```diff
@@ -25,11 +25,15 @@ def _get_url_size(fname, **open_kwargs):
     return size
 
 
-def _copy_btw_filesystems(input_opener, output_opener, BLOCK_SIZE=10_000_000):
+def _copy_btw_filesystems(input_opener, output_opener, BLOCK_SIZE=10_000_000, **kwargs):
+    streaming = True if kwargs.pop("block_size", True) == 0 else False
+    if os.getenv("PANGEO_FORGE_BLOCK_SIZE"):
+        BLOCK_SIZE = int(os.getenv("PANGEO_FORGE_BLOCK_SIZE"))
```

Comment:
This needs to be documented in the documentation. Perhaps we need a page on environment variables?

Your PR description makes it sound like `PANGEO_FORGE_BLOCK_SIZE` only governs logging behavior, but here is seems like it changes the actual block size for reads.

Reply:
> This needs to be documented in the documentation. Perhaps we need a page on environment variables?

👍 

> Your PR description makes it sound like PANGEO_FORGE_BLOCK_SIZE only governs logging behavior, but here is seems like it changes the actual block size for reads.

In the case of range requests, yes, it would theoretically do that, though I'd need to test it more in that context to confirm. But in the case of streaming downloads, my understanding is `fsspec_open_kwargs = {"block_size": 0}` actually forces
```python
data = source.read(BLOCK_SIZE)
```
to ignore `BLOCK_SIZE` in favor of many smaller streaming reads. So for this PR, the notion I'm trying to capture in the env variable is perhaps closer to `PANGEO_FORGE_LOGGING_INTERVAL_FOR_STREAMING_READS`, but that seemed like an oddly specific concept to convey in an env variable (even if it's name were abbreviated), so I ended up here.

In this form, it does have the potential to change the block size for range request reads, but as I've imagined it thus far, that use case is not intended. (If we keep some form of this implementation, that could be noted in the docs.) As for how it actually works for changing the logging interval, I'll address that in the answer to your question below.